README.md

```diff
@@ -406,6 +406,70 @@ Loans include information from the item, including locations, holdingsRecordId a
 
 Requests include information from the item, including holdingsRecordId and instanceId.
 
+#### Hold shelf clearance report CSV
+
+To create hold expiration report that can be used by staff to clear expired and cancelled holds from the shelf and put them back into circulation.
+Then generate a report based on this logic:
+* Find all of the items that are `Awaiting pickup`
+* Choose the item’s which either have an empty request queue or request queue doesn't contain request with status `Open - Awaiting pickup`
+* Select the request for each item that expired or was cancelled on the hold shelf most recently (by ranking the closed requests by a date used only for this purpose)
+* Choose only the requests where the `pickup service point` matches the chosen service point
+
+The API for generating a report is based on the presence of the 'awaitingPickupRequestClosedDate' property for the request JSONB.
+Such behavior is required by database trigger for request update in the mod-circulation-storage
+See [CIRCSTORE-127](https://issues.folio.org/browse/CIRCSTORE-127)
+
+Data involved in the formation of the report:
+* Requester name: lastName, firstName
+* Requester barcode
+* Item title
+* Item barcode
+* Call number
+* Request status
+* Hold shelf expiration date: Date with timestamp
+
+#### Example Request
+
+```
+GET http://{okapi-location}/circulation/requests-reports/hold-shelf-clearance/:{servicePointId}
+```
+
+#### Example Success Response
+
+```
+HTTP/1.1 200 Ok
+content-type: application/json; charset=utf-8
+content-length: 230
+
+{
+  "requests": [
```

Comment:
Requests included in this report were fulfilled to the hold shelf at a service point, meaning that all of them should have a `pickupServicePointId`.

Reply:
Added `pickupServicePointId` to the example of a request

Reply:
It appears the `pickupServicePoint` object has been added, and not the `pickupServicePointId`?

Reply:
Is the service point fetched during the report?

Reply:
I tested it again and modified the README. Only the following is added to the request: pickupServicePointId, awaitingPickupRequestClosedDate (new field)

Reply:
Fixed